### ISSUE TITLE
iOS export versions and product name not correctly set in generated XCode project

### ISSUE THREAD

**Issue by lostminds** [godotengine/godot#70744]
### Godot version

4.0beta10

### System information

macOS 13.0.1

### Issue description

When exporting to iOS Godot generates an Xcode project you then need to run in Xcode (and often make a few last adjustments to) to test or publish your game on iOS. However, it seems that the version and product name is not configured correctly in the XCode project. If you open the target identity tab in the exported project you can see that only the bundle identifier is filled in, even though you've supplied both the versions and the game name in Godot.

![Screenshot 2022-12-30 at 16 45 33](https://user-images.githubusercontent.com/17763524/210088914-7081278c-4adc-4065-9126-86fbc02749df.jpg)

The reason for this is I think that these details are only filled in in the product info.plist, which one might consider correct as it's what is used in the exported final bundle. However, this does not seem like the way Xcode prefers this information to be stored, and therefor it is not displayed or used correctly in Xcode.

Instead, the preferred way to store this info for Xcode projects ([as indicated here](https://developer.apple.com/documentation/bundleresources/information_property_list/managing_your_app_s_information_property_list?language=objc)) now seems to be to configure build settings for these properties in the project.pbxproj file (inside the .xcodeproj bundle). Namely the build settings "Bundle Display Name" (Display name), "Current Project Version" (Build) and "Marketing Version" (Version). These are then the properties used by XCode in the target identity interface etc. (And yes, this is yet another set of dictionary keys with different names for this same information, different from the interface labels and info.plist keys)

To make sure the correct info is used in the exported plist placeholders for these build settings can then be used in the plist to use the values from the build settings. Here's an example from another macOS project of mine, with these presets generated by XCode:

![Screenshot 2022-12-30 at 16 46 40](https://user-images.githubusercontent.com/17763524/210090016-a364981a-0b49-4746-85c4-0e14ceb44c6a.jpg)

In other words XCode seems to assume these settings are defined and read from build settings and then written to the info.plist when building instead of being stored in the info.plist as godot does now.

### Steps to reproduce

Set up project export for iOS with versions defined and a project name from project settings
Export for iOS
Open the exported project in XCode and see that in the target info the identity panel fields for product name and versions are empty

### Minimal reproduction project

This should be the same for all iOS exports

**Comment by HEAVYPOLY** [comment]
Same issue happening in current 3.x branch

**Comment by bruvzg** [comment]
We do it for bundle identifier, so why not. Itš not a big change - https://github.com/godotengine/godot/pull/71143
